app/services/openlibrary_service.rb

```diff
@@ -0,0 +1,33 @@
+class OpenlibraryService
+  include HTTParty
+  base_uri 'https://openlibrary.org/api'
+
+  default_params format: :json, jscmd: 'data'
+
+  class BookNotFoundError < StandardError
+  end
+
+  def search_by_isbn(isbn)
+    isbn_string = "ISBN:#{isbn}"
+    options = { query: { bibkeys: isbn_string } }
+    response = self.class.get('/books', options)
+
+    return parse_response_book(response[isbn_string], isbn) if response.success?
+
+    raise BookNotFoundError, response.response
+  end
+
+  private
+
+  def parse_response_book(book, isbn)
+    {
+      isbn: isbn,
+      title: book['title'],
+      subtitle: book['subtitle'],
+      number_of_pages: book['number_of_pages'],
+      authors: book['authors'].map { |author| author['name'] }
+    }
+  rescue StandardError
```

Comment:
Why you are doing this rescue?

Reply:
If `book['authors']` is `nil` or not an array it throws NoMethodError

Reply:
```ruby
def parse_response_book(book, isbn)
    raise CustomErrors::BookNotFoundError, 'Invalid response book' unless
      book&.include_all?('title', 'subtitle', 'number_of_pages', 'authors') &&
      book['authors'].is_a?(Array)

    {
      isbn: isbn,
      title: book['title'],
      subtitle: book['subtitle'],
      number_of_pages: book['number_of_pages'],
      authors: book['authors'].map { |author| author['name'] }
    }
  end
```